packages/web-components/src/drawer/drawer.ts

```diff
@@ -0,0 +1,279 @@
+ * @method hide() - Closes the drawer if it is currently open.
+ * @method clickHandler(event) - Handles click events on the drawer.
+ * @method keydownHandler(event) - Handles keydown events on the drawer.
+ * @method openChanged(oldValue, newValue) - Handles changes to the `open` property.
+ * @method typeChanged(oldValue, newValue) - Handles changes to the `type` attribute.
+ *
+ * @summary A flexible drawer component that can be used in various configurations such as modal, non-modal, alert, inline, overlay, with different sizes and positions.
+ *
+ * @tag fluent-drawer
+ */
+
+export class Drawer extends FASTElement {
+  /**
+   * This method is called when the custom element becomes connected to the document's DOM.
+   * @public
+   */
+  public connectedCallback(): void {
+    super.connectedCallback();
+    /**
+     * By placing the syncDialogOpenState() inside Updates.enqueue(), we ensure that any actions to synchronize the component's state based on initial attributes are deferred until after all other pending DOM updates are processed.
+     * This ensures that the internal state of the dialog element aligns with the initial attributes specified on the <fluent-drawer>, such as automatically opening the dialog when the 'open' attribute is present at the time of connection to the DOM.
+     */
+    Updates.enqueue(() => {
+      this.syncDialogOpenState();
+    });
+  }
+
+  /**
+   * This method is called when the custom element is disconnected from the document's DOM.
+   * @public
+   */
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+  }
+
+  /**
+   * The dialog element.
+   * @public
+   *
+   */
+  @observable
+  public dialog!: HTMLDialogElement;
+
+  /**
+   * Determines whether the drawer should be displayed as modal, non-modal, or alert.
+   * When in modal or alert mode, an overlay is applied over the rest of the view.
+   * @public
+   */
+  @attr({ attribute: 'modal-type' })
+  public modalType: DrawerModalType = DrawerModalType.modal;
+
+  /**
+   * The ID of the element that labels the drawer.
+   * @public
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  public ariaLabelledby?: string;
+
+  /**
+   * The ID of the element that describes the drawer.
+   * @public
+   */
+  @attr({ attribute: 'aria-describedby' })
+  public ariaDescribedby?: string;
+
+  /**
+   * Sets the type of the drawer (overlay/inline).
+   * @public
+   * @defaultValue overlay
+   */
+  @attr
+  public type?: DrawerType;
+
+  /**
+   * Sets the position of the drawer (start/end).
+   * @public
+   * @defaultValue start
+   */
+  @attr
+  public position?: DrawerPosition;
+
+  /**
+   * Sets the size of the drawer (small/medium/large).
+   * @public
+   * @defaultValue medium
+   */
+  @attr({ attribute: 'size' })
+  public size?: DrawerSize;
+
+  /**
+   * Sets the open state of the drawer
+   * @public
+   * @defaultValue false
+   */
+  @attr({ mode: 'boolean' })
+  public open: boolean = false;
```

Comment:
With the API to show/hide, I'm not sure we need a declarative way to set "open" and need to manage this. Let's look to align with the latest proposed `dialog` implementation.